simapp/sim_test.go

```diff
@@ -45,120 +44,186 @@ func init() {
+	sims.Run(t, NewSimApp, setupStateFactory, func(t *testing.T, ti sims.TestInstance[*SimApp]) {
+		app := ti.App
+		t.Log("exporting genesis...\n")
+		exported, err := app.ExportAppStateAndValidators(false, []string{}, []string{})
+		require.NoError(t, err)
+
+		t.Log("importing genesis...\n")
+		newTestInstance := sims.NewSimulationAppInstance(t, ti.Cfg, NewSimApp)
+		newApp := newTestInstance.App
+		var genesisState GenesisState
+		require.NoError(t, json.Unmarshal(exported.AppState, &genesisState))
+		ctxB := newApp.NewContextLegacy(true, cmtproto.Header{Height: app.LastBlockHeight()})
+		_, err = newApp.ModuleManager.InitGenesis(ctxB, genesisState)
+		if IsEmptyValidatorSetErr(err) {
+			t.Skip("Skipping simulation as all validators have been unbonded")
+			return
+		}
+		require.NoError(t, err)
+		err = newApp.StoreConsensusParams(ctxB, exported.ConsensusParams)
+		require.NoError(t, err)
+
+		t.Log("comparing stores...")
+		// skip certain prefixes
+		skipPrefixes := map[string][][]byte{
+			stakingtypes.StoreKey: {
+				stakingtypes.UnbondingQueueKey, stakingtypes.RedelegationQueueKey, stakingtypes.ValidatorQueueKey,
+				stakingtypes.HistoricalInfoKey, stakingtypes.UnbondingIDKey, stakingtypes.UnbondingIndexKey,
+				stakingtypes.UnbondingTypeKey,
+			},
+			authzkeeper.StoreKey:   {authzkeeper.GrantQueuePrefix},
+			feegrant.StoreKey:      {feegrant.FeeAllowanceQueueKeyPrefix},
+			slashingtypes.StoreKey: {slashingtypes.ValidatorMissedBlockBitmapKeyPrefix},
+		}
+		AssertEqualStores(t, app, newApp, app.SimulationManager().StoreDecoders, skipPrefixes)
+	})
+}
 
-	// Run randomized simulation
-	_, simParams, simErr := simulation.SimulateFromSeed(
-		t,
-		os.Stdout,
-		app.BaseApp,
-		simtestutil.AppStateFn(app.AppCodec(), app.AuthKeeper.AddressCodec(), app.StakingKeeper.ValidatorAddressCodec(), app.SimulationManager(), app.DefaultGenesis()),
-		simtypes.RandomAccounts, // Replace with own random account function if using keys other than secp256k1
-		simtestutil.SimulationOperations(app, app.AppCodec(), config),
-		BlockedAddresses(),
-		config,
-		app.AppCodec(),
-		codectestutil.CodecOptions{}.GetAddressCodec(),
-	)
+// Scenario:
+//
+//	Start a fresh node and run n blocks, export state
+//	set up a new node instance, Init chain from exported genesis
+//	run new instance for n blocks
+func TestAppSimulationAfterImport(t *testing.T) {
+	sims.Run(t, NewSimApp, setupStateFactory, func(t *testing.T, ti sims.TestInstance[*SimApp]) {
+		app := ti.App
+		t.Log("exporting genesis...\n")
+		exported, err := app.ExportAppStateAndValidators(false, []string{}, []string{})
+		require.NoError(t, err)
+
+		t.Log("importing genesis...\n")
+		newTestInstance := sims.NewSimulationAppInstance(t, ti.Cfg, NewSimApp)
+		newApp := newTestInstance.App
+		_, err = newApp.InitChain(&abci.InitChainRequest{
+			AppStateBytes: exported.AppState,
+			ChainId:       sims.SimAppChainID,
+		})
+		if IsEmptyValidatorSetErr(err) {
```

Comment:
There are multiple seeds executed. Some may cause all the validators to undelegate/ jail. In fuzz tests, this is not necessarily a failure scenario. We accept that the test run is not realistic and can not be completed as defined. We just skip this seed and move on. It would only be a problem, when all seed get into this state.